### PR TITLE
Fix for https://issues.redhat.com/browse/JBCS-1261

### PR DESCRIPTION
**Pull request by jfclere**
Terminate child process that have been started by the service.
That fixes problem what using wildfly standalone.bat that creates java.exe, if the stop cli fails stopping the bat doesn't stop the java.exe...

### CODE REVIEW

File: src/native/windows/src/rprocess.c [hunk 1]
```diff
@@ -715,12 +792,14 @@ apxProcessWait(APXHANDLE hProcess, DWORD dwMilliseconds, BOOL bKill)
     if (hProcess->dwType != APXHANDLE_TYPE_PROCESS)
         return WAIT_ABANDONED;
 
+    apxLogWrite(APXLOG_MARK_DEBUG "apxProcessWait..");
```
Comment (garydgregory) [inline src/native/windows/src/rprocess.c:795]: Typo: extra dot.

**Comment by garydgregory** [comment]
@jfclere 
Thank you for your PR.  I like the extra logging. 
@markt-asf how do this PR look to you?

File: src/native/windows/src/rprocess.c (hunk 1 above)
Comment (jfclere) [reply src/native/windows/src/rprocess.c]: Fixed.

File: src/native/windows/src/rprocess.c [hunk 2]
```diff
@@ -316,13 +373,16 @@ static BOOL __apxProcessClose(APXHANDLE hProcess)
     lpProc = APXHANDLE_DATA(hProcess);
     CHECK_IF_ACTIVE(lpProc);
 
+    __apxProcessTerminateChild(lpProc->stProcInfo.dwProcessId, TRUE);
     /* Try to close the child's stdin first */
```
Comment (markt-asf) [inline src/native/windows/src/rprocess.c:377]: Can you explain the purpose of the above call? It seems odd to be forcibly terminating child processes this early. What am I missing?

Comment (jfclere) [reply src/native/windows/src/rprocess.c:377]: Mixing TRUE and FALSE probably...
https://github.com/jfclere/commons-daemon/blob/master/src/native/windows/src/rprocess.c#L413
Here the process MUST be terminated not in line 376.
Good catch I will fix and retest on Monday.

Comment (markt-asf) [reply src/native/windows/src/rprocess.c:377]: Got it. It was `dryrun` being `TRUE` that I missed. This makes sense now. It generates a debug log of the current child processes before a normal shutdown. The forced shutdown only happens if the normal one goes wrong. LGTM.

Comment (jfclere) [reply src/native/windows/src/rprocess.c:377]: Cool should I add a comment in the code to explain it traces the process tree and another where it kills the process tree?

Comment (markt-asf) [reply src/native/windows/src/rprocess.c:377]: It was obvious once you explained it but a comment wouldn't hurt.

Comment (jfclere) [reply src/native/windows/src/rprocess.c:378]: Done  Many thanks for the review.

Comment (garydgregory) [reply src/native/windows/src/rprocess.c:378]: > Cool should I add a comment in the code to explain it traces the process tree and another where it kills the process tree?

👍 for comments.

**Pull request merged**